### ISSUE TITLE
[Bug] monaco monarch embedded language examples not behaving correctly

### ISSUE THREAD

**Issue by frankli0324** [microsoft/monaco-editor#4075]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Link

https://microsoft.github.io/monaco-editor/playground.html?source=v0.40.0#XQAAAALpGwAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscxyX4PBYlrMCleTc0f2p3Snm_TA7RYQPdPqtK-6IkeTGFd19Enemr8s4kTpQxV2aoMGSUSUfzOvYz0V91IOHrL2bWgKFPl1i_I8_8trkQtViPwdtAseFLaG9myNe7cHR0KUBezeb96T6WuTFf21xJYbDaDJQcs5TnIynQ1gFxqkEUE8TlG4hyNEoy_2fyB02TCRnvVJkorYMkTh5MvAE7UR8Wu_WxzPepg_mwXdpUApakzw1iDftvJPFKOQ80G08nMFZoIyish9-iuAkruicegFisRUu3--__9RZRSGZWXb_gqeUCxWubyZQPhEitbtRQV9CEJYdSVzNsRLAWI0jyuQ9ApH72uBnUf7hgQqtQPzSZeZJj5mwm2lsoL4_7SADCpwfhm4nBiXQ490M99mNqT1qJgtXMYUZHNplqhLsLZAwuu-l01xKYdZ-JcoQZ7mBigGgg82TGwXN-xvMMrWPttHMRW07PY5HXBWEdnrNpGSCsBn7zJzESEtqDH1-Vz5HydY-RO1d5JG8IFzoLfv9FFDrlZrvhner4rMGWwdcNLUp4fFFksqx-ehepkQtROAMlFtqcHrccIqO9z1O4H0BjPjD-27KM0tCjmApFU39yAe-5-JJQ4IdG5hVK0Nva642wdIHd4pBmSXF5RGHxI_Ny28Q9dFeAmkYh7TQAiJfZB8bqrMa_7X2iTp3SRdWT9WQKnciMZCKtCUOW_0kveYFxUGWlJQF7rNwEcbiKitz6ID-e3VMUx2VjN4L7yzGkI_Z7Wb4pg2rTUJNIsoNv4QQhsi3_HQTUUpdEWhR5nT3hS_VRG7sY2e4HtBXjMXtSWgazFtANZKMUjN8wZuBfqDQmmOvhOnAUYOPIPWr2Dl4ox6KrG_2aoKCgE8-gIMoH4__5KuPS668nAIdOIRXeoxbMA7f_LIqNaEYjkVLnqs8yDygUd8528lcuRITpr8YsKblrXzaml-5E7fPrFQghYSQ6QMhVhrHFJTRPKEW-pRD38FYu_4vH-ThnibuyaVhPLvWrMXywMntNZiptMrNOyPphG31xYHmP_69cewu0MJbCGmKc06-DKMuwPJR_S5U5Z7BRqm45zAgSWRTgCjY24kBSuO0yi-qE7l6gxOZJE05NAmZchuMWDlVXo2bHzJ2aF_ALpHVxNDT3LXYAgFpezbB7lk_vp26SlZeKAAYWhtY-A4A0kpVyR1kkuc-kqYrc1SJslvv5nh231ks20bNk8uHY5leNUkub3KQAxroW9mEn3wB8c75DCwITPV-_gMUsk-aTHHcEDMhS7X8i3fRmqUaz4f13NV23t3McaNR4uRZMhO4MSajZSw9TWYIN_n4AQkFABGvSUCp7TjzHIuZWsG8gDrFjQk6KVuR20PBMNpPA-asUzRVhdT3czdTxnVpJWCVo7XjsfJea1GCtG4Mz6U1q-IKh_3SQxRtKlsZk8mT8mfSidwQpt37RPeW-mUvzXul-ITRJKECf0aMbeE6-WUuGVdkmDcCt2aLNhmLGfjHwY5vn0aIM2CWpzJ0uQodVJ_L7SiP7F4D2jlr-SmBkl-gh_8HoJ3g6EgISXP8OYujRsfhnWDr_6l33V6Bp46ErSjiPS6IpieHFAlU3l5qKv9v3n9sDeZd2yeKWIsBBClNgo79yi_XCYSF1_x-iyDyvV1iB-AGcoc2E4QTAZDvf_nlCAcGKaLxdx6xK_JiAiMv-ndB69ydjbq3OHHM8GkWpvjeTH24bGqQVdBqu5zC7pMnxW-_4s83zJlCN-tu-R-N_DSbw0HM11Zeey4aUbRpK_SEgXvZfBtCiD7Oh5o5KpJe4QpaHL7dmrbSq0eg6PYHyDvvKJ21qSdXSg1tMJ2W474w_YQBg00mEdmBtSwQ-xFm0yc7VcDKTlRTvWqi9mLyB-e1Y1A16FH4V39pWR8e8CGnNHjjzCfYdhFWTUMcmtXGV4v_uF_VWz3bmyPoUfu7KL-Npkts8HiJGk6bTASoqjXzseZzVSAqmqwf9vQyXVVmNosyOVvL5eakpFxTftSaRmlgjuydxgRfWEHrOKDNJiX68kGoZUyOJ-vPQ0vPYiRNQkMEfilePLI1JP03nTMhTeCsWPKuygMKInEr4FZB5OVwuA1Njks6UPkbWUjXFTyDrQr6jJabxf6KTdhnqrOZd9GJug9kzQGJxFyJozpBIykvAYaZlUNumKRWzctUOUs3ChQ5qJ2Zj55F_xzOZEuApDgjVcyfI5XHCqA0mCnl9Dn2sw8FRopUH5rjaleD4XyXWYhMdW-GGV0HRgFGQRElK-e4wuQf77c1Y

### Reproduction Steps

see https://microsoft.github.io/monaco-editor/monarch.html html and markdown examples

### Actual (Problematic) Behavior

![image](https://github.com/microsoft/monaco-editor/assets/20221896/1867adda-f80e-404c-9dba-22db1f7e2b27)

### Expected Behavior

javascript code be correctly rendered

**Comment by SanjayManjunath** [comment]
hey @frankli0324, i checked the link provided. Can you please confirm what ive understood is correct?   "Hi </script>" must be displayed on the webpage?

**Comment by frankli0324** [comment]
@SanjayManjunath no, I mean it should be correctly tokenized and highlighted as javascript string. for example:
![image](https://github.com/microsoft/monaco-editor/assets/20221896/04509199-b3d0-43ab-b433-833d4daa774f)

**Comment by frankli0324** [comment]
@SanjayManjunath I think that's off-topic. please consult search engines for starting and debugging javascript projects. You could, however, start with [vite](https://vitejs.dev/guide/).
